src/index.js

```diff
@@ -471,12 +474,131 @@ class XRState {
 }
 const xrState = GlobalContext.xrState = new XRState();
 
-if (nativeBindings.nativeVr) {
+if (nativeBindings.oculusVr) {
+  nativeBindings.nativeVr.requestPresent = function (layers) {
+    const layer = layers.find(layer => layer && layer.source && layer.source.tagName === 'CANVAS');
+    if (layer) {
+      const canvas = layer.source;
+      if (!vrPresentState.glContext || nativeBindings.oculusVr) {
+        let context = canvas._context;
+        if (!(context && context.constructor && context.constructor.name === 'WebGLRenderingContext')) {
+          context = canvas.getContext('webgl');
+        }
+        const window = canvas.ownerDocument.defaultView;
+
+        const windowHandle = context.getWindowHandle();
+        nativeBindings.nativeWindow.setCurrentWindowContext(windowHandle);
+
+        // fps = VR_FPS;
+
+        const system = vrPresentState.system = vrPresentState.system || nativeBindings.oculusVr.Oculus_Init();
+        const lmContext = vrPresentState.lmContext || (nativeBindings.nativeLm && new nativeBindings.nativeLm());
+
+        system.SetupSwapChain(contexts[0]);
+        const {width: halfWidth, height} = system.GetRecommendedRenderTargetSize();
+
+        const maxTextureSize = 4096;
```

Comment:
I think this was updated upstream to a different behavior for openvr. We should probably be consistent both ways, so maybe this should be broken out into a helper function.

More generally, all of this bootstrapping seems like it could be parameterized by the hardware API and most of this code could be shared between the VR platforms.